tests/bluetooth/host/buf/bt_buf_get_rx/src/main.c

```diff
@@ -38,13 +38,13 @@ void test_returns_null_type_bt_buf_evt(void)
 		memory_pool = bt_buf_get_hci_rx_pool();
 	}
 
-	ztest_expect_value(net_buf_alloc_fixed, pool, memory_pool);
-	ztest_returns_value(net_buf_alloc_fixed, NULL);
-
-	ztest_expect_value(net_buf_validate_timeout_value_mock, value, timeout.ticks);
+	net_buf_alloc_fixed_fake.return_val = NULL;
 
 	returned_buf = bt_buf_get_rx(BT_BUF_EVT, timeout);
 
+	validate_net_buf_alloc_called_behaviour(memory_pool, &timeout);
```

Comment:
I wonder if this test is too restrictive. Why is it wrong for `bt_buf_get_rx` to result in multiple calls to `net_buf_alloc`? There must be a rationale for this restriction.

Reply:
Without something describing the expected behavior, I'm extracting test cases from the current implementation.
Currently, a single call to `bt_buf_get_rx()` should result in a single call to `net_buf_alloc().`

When the implementation changes, the test should be adapted as well